### ISSUE TITLE
Caused by: org.apache.coyote.http11.AbstractHttp11Processor.process Error processing request  java.lang.NoSuchFieldError: attributes

### ISSUE THREAD

**Issue by nhsoft** [magro/memcached-session-manager#301]
download the below jars and put them in $CATALINA_HOME/lib/, jars list：
1. asm-5.1.jar
2. kryo-3.0.3.jar
3. kryo-serializers-0.38.jar
4. memcached-session-manager-1.9.2.jar
5. memcached-session-manager-tc8-1.9.2.jar
6. minlog-1.3.0.jar
7. msm-kryo-serializer-1.9.2.jar
8. reflectasm-1.11.3.jar
9. spymemcached-2.11.1.jar
10. objenesis-2.1.jar

tomcat context.xml config detail

> <Manager className="de.javakaffee.web.msm.MemcachedBackupSessionManager"
		memcachedNodes="c9cf3bdc2455b.m.cnhzaliqshpub001.ocs.aliyuncs.com:11211"
		username="c9cf3bdc2455b" 
		password="*******"
		memcachedProtocol="binary"
		sticky="false"
		sessionBackupAsync="false"
		requestUriIgnorePattern=".*\.(ico|png|gif|jpg|css)$"
		transcoderFactoryClass="de.javakaffee.web.msm.serializer.kryo.KryoTranscoderFactory"/>

exception logs with running tomcat8 ==============>>>

> 03-Jun-2016 11:16:18.690 SEVERE [http-nio-80-exec-9] org.apache.coyote.http11.AbstractHttp11Processor.process Error processing request
>  java.lang.NoSuchFieldError: attributes
> 	at de.javakaffee.web.msm.MemcachedBackupSession.getAttributesFiltered(MemcachedBackupSession.java:522)
> 	at de.javakaffee.web.msm.BackupSessionTask.call(BackupSessionTask.java:109)
> 	at de.javakaffee.web.msm.BackupSessionTask.call(BackupSessionTask.java:51)
> 	at de.javakaffee.web.msm.BackupSessionService$SynchronousExecutorService.submit(BackupSessionService.java:346)
> 	at de.javakaffee.web.msm.BackupSessionService.backupSession(BackupSessionService.java:205)
> 	at de.javakaffee.web.msm.MemcachedSessionService.backupSession(MemcachedSessionService.java:1086)
> 	at de.javakaffee.web.msm.RequestTrackingHostValve.backupSession(RequestTrackingHostValve.java:230)
> 	at de.javakaffee.web.msm.RequestTrackingHostValve.invoke(RequestTrackingHostValve.java:159)
> 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
> 	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616)
> 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
> 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:528)
> 	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1099)
> 	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:672)
> 	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1520)
> 	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1476)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
> 	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
> 	at java.lang.Thread.run(Unknown Source)

**Comment by magro** [comment]
This was also reported in #297. Which version of tomcat are you using?

Btw, as said in SetupAndConfiguration serialization jars (msm-kryo-serializer and all transitive dependencies) should go into WEB-INF/lib/ instead of $CATALINA_HOME/lib/.

**Comment by magro** [comment]
I just released msm 1.9.3 with the fix for this.

**Comment by nhsoft** [comment]
Yes, After upgrade to 1.9.3， the issue fixed, it works for me.

**Comment by magro** [comment]
Great!

**Comment by danpipe** [comment]
Hello @magro, I appreciate this is a very old post I'm commenting on but if you see this comment I would much appreciate your insight, as we have just started seeing this issue.

Our project uses an ecommerce platform called Hybris, which runs on the Tomcat app server, and we are using MSM. After a recent upgrade of the Hybris version, we have started seeing the above exception in the logs, and I cannot understand why.

The Tomcat version was 7.0.59 before the upgrade, and is now 7.0.100.

We are using the following MSM libs:

memcached-session-manager-1.8.3.jar
memcached-session-manager-tc7-1.8.3.jar

I see that your fix for this issue in release 1.9.3 was to handle differences between Tomcat 7 and earlier versions - but we are seeing the issue despite Tomcat remaining on v7.

- Should we expect to be seeing this issue when using MSM 1.8.3 on Tomcat 7?
- Would upgrading to MSM 1.9.3 resolve the issue regardless?
- Any idea why we would start seeing this issue now?

Many thanks in advance, 
Dan
